### ISSUE TITLE
PF drivers from initial conditions

### ISSUE THREAD

**Issue by pek-neos** [Yellow-Dog-Man/Resonite-Issues#1072]
### Describe the bug?

PF drives do not start from the initial conditions in the inspector as expected by the user.

### To Reproduce

no

### Expected behavior

Would it be possible for the PF drives to start from the initial conditions in the inspector and not reset them to zero?

### Resonite Version Number

last

### What Platforms does this occur on?

Windows

### What headset if any do you use?

hp reverb g2

### Log Files

no

**Comment by Banane9** [comment]
Drives set the value to whatever their input is - by default this is, well, the default for the type, so 0, false, null, or equivalents.
They don't care about the value already in the field, as such your expectation is just wrong 😅

**Comment by pek-neos** [comment]
Yes, thank you, that is exactly what I want to change. So that, for example, objects in the scene move from where they are right now and not from zero position. That they change the color from the one they currently have and not from white.

**Comment by JackTheFoxOtter** [comment]
That's not how drives are supposed to work in FrooxEngine. When a field is driven, that means its value is coming from whatever the drive source is. With ProtoFlux you can construct a drive that takes whatever value previously was in the target field and overwrite it with the result of some operation that takes the previous value as an input, but you have explicitly write that evaluation to do so. IIRC the globe in the default cloud home is an example of that.

**Comment by Banane9** [comment]
> Yes, thank you, that is exactly what I want to change. So that, for example, objects in the scene move from where they are right now and not from zero position. That they change the color from the one they currently have and not from white.

As Jack described, you can achieve that just fine - but unless you specifically set it up, it makes no sense for drives to work that way.

**Comment by Frooxius** [comment]
Just to add context for this, I've asked pek (he's a university teacher using Resonite in his class and has been supporting it for a long while) to make this issue so it can be tracked.

The idea is that when you start driving something, it creates the input node with the current value of the driven field.

That way it doesn't reset the value right away, but just moves it elsewhere.

**Comment by Banane9** [comment]
> The idea is that when you start driving something, it creates the input node with the current value of the driven field.
> 
> That way it doesn't reset the value right away, but just moves it elsewhere.

That makes more sense, but was absolutely unclear from the description and ellaborations 😅

**Comment by JackTheFoxOtter** [comment]
Ah! Okay, yeah that is a pretty good idea! Apologies for misunderstanding the intention.

**Comment by pek-neos** [comment]
Thank you very much for the comments and solution. Please excuse me for the unclear description.
